### ISSUE TITLE
High Albedo bias in SP mode

### ISSUE THREAD

**Issue by rosiealice** [NGEET/fates#837]
From comparisons with iLAMB, it appears that FATES-SP has a high albedo and reflected shortwave bias. 

We had originally thought that this could be tracked to the layering of the surface radiation scheme. Indeed, some fraction (a bit less than half) of this can be fixed by moving from a scheme with thick layers to one with thinner layers (see fig below where we move from the CLM5 comparison run (SP, with various modifications) to the 0.2 layer thickness FATES (big increase in reflected radiation, FSR), to variable thickness FATES with thin layers (not much change) to FATES with variable thick layers (a further large increase in reflected radiation). 
![SP_prog_FSR_10 (2)](https://user-images.githubusercontent.com/12954691/154101919-6ffe3c14-98b2-44c5-944b-985459d07465.png)

but even with thin layers, there is a relatively large difference. 
The good news is that the run time is so dominated by history variable machinations at this point that changing from the thin to the thick layering scheme barely changed the total cost ;)

There are three avenues to pursue:
1. making very very very thin layers and seeing if there is any configuration of layering where this goes away. 
2. Looking at comparison runs of CLM5 and FATES-SP with NO STEM AREA. The error tracks a little with the ESAI, and there may be some critical differences in how the layering scheme comprehends SAI now compared to big leaf. 
3. Look at whether these errors are within the parametric tolerance of the leaf and stem reflectance parameters and might be a tuning issue. 
 
Here are the changes in SABV and SABG (veg and ground absorbed solar) for completeness
![SP_prog_SABG_10 (1)](https://user-images.githubusercontent.com/12954691/154101891-11a5316f-076b-4ff7-a936-3c6676d9d0cf.png)

![SP_prog_SABV_10 (1)](https://user-images.githubusercontent.com/12954691/154101880-12121ad7-2e9d-465b-939e-2296a555694e.png)

 Noting that all of these figures are for the month of June, to speed up making these plots!

**Comment by rosiealice** [comment]
ILAMB comparisons here
https://webext.cgd.ucar.edu/I2000/FATES_SP_4x5test/lnd/_build_FATESSP_FATESnocomp/RadiationandEnergyCycle/Albedo/GEWEX.SRB/GEWEX.SRB.html

**Comment by rosiealice** [comment]
Looking at explanation #2 illustrates something a little wierd, which is that the reflectance responds in the opposite direction to the removal of SAI in FATES and CLM5. 
![SP_prog_FSR_11](https://user-images.githubusercontent.com/12954691/154085472-bd14c8c0-3a43-4a99-80e2-772136eb7108.png)

In both models, removing SAI reduces the amount of radiation absorbed by the canopy. 
![SP_prog_SABV_11](https://user-images.githubusercontent.com/12954691/154085638-5e362e3b-17cd-4e55-a051-f0069a4f195a.png)

and increases the amount hitting the ground. 
![SP_prog_SABG_11](https://user-images.githubusercontent.com/12954691/154085734-8797290a-74e3-46bc-9856-a479c6146dc7.png)

but in FATES removing stems reduces the reflected radiation. (So adding stems increases the reflectence. So the leaves+stems are 'shiner' than the leaves on their own). 

This comparison of FATES with and without stems is with the thicker (1,0LAI) leaf layers. I will try it again with thinner layers to check it's not a feature of the grainy canopy.

**Comment by rosiealice** [comment]
Returning to the problem of the SAI and how it impacts FSR. I tried this with a high vertical resolution canopy and the results were similar.

![SP_prog_FSR_11 (1)](https://user-images.githubusercontent.com/12954691/154096851-42df3448-f201-48e5-9973-e0b7d505fa81.png)

 So adding stem area increases the light absorbed by the canopy, but doesn't impact the reflected fraction in FATES. 
![SP_prog_SABV_11 (1)](https://user-images.githubusercontent.com/12954691/154097532-425c9719-e673-475d-b273-e51ef6d6183f.png)

![SP_prog_SABG_11 (1)](https://user-images.githubusercontent.com/12954691/154097547-1dd85a40-46cc-414b-af58-388d32146c9a.png)

**Comment by rosiealice** [comment]
### Impact of increase factor on fluxes. 
Holding the width of the top bin at 0.1, this is the impact of changing the increase factor from 1.5 to 1.75 to 2.0. There is a small (order 1 watt) change from 1,5 to 1.75. 

![SP_prog_FSR_13 (2)](https://user-images.githubusercontent.com/12954691/154115453-ddeac038-efd5-42ce-9868-66227cf8385d.png)

![SP_prog_SABV_13 (2)](https://user-images.githubusercontent.com/12954691/154115570-98719a4d-420e-4db2-ba36-b607ef1804af.png)

**Comment by ckoven** [comment]
@rosiealice what about 0.2 top-bin resolution?

**Comment by rosiealice** [comment]
You mean as the new default? With 1,5 as the increase factor? I think that should get rid of most of the layering induced biases...

**Comment by aswann** [comment]
My question might be way too naive, but are the NIR and VIS reflectances set the same for stems in both FATES and CLM? (I'm sure you thought of that already, but figured I'd ask anyway). I know @marysa found the NIR reflectances had a big influence on total albedo.

**Comment by rosiealice** [comment]
@walkeranthonyp I was wondering whether your MAAT implementation of th Norman RTM might be an interesting way to baseline what should be happening here under some controlled condition?

**Comment by ckoven** [comment]
@rosiealice I was just suggesting a comparison of 0.2 top-level bin width alongside the ones that you've already done at 0.1, 0.5, and 1.0

**Comment by rosiealice** [comment]
OK right. Yes, can do that for sure.

**Comment by rosiealice** [comment]
Hi Anthony, 
We should probably set up the details of this offline, but yes, it would be good to test the albedo predictions for a standrardized setup in terms of LAI, layering, solar angle etc.

There are two issues. The first is the bias in the albedo in general, and the second is the response to adding SAI. For the first, we can consider a 'leaf only' setup, since there is a mismatch with CLM5 for that as well as the condition with stems. 

One thing is that, given the iterative solution, we might not be able to track down the nature of the biases without doing some relatively detailed interrogation of how, .e.g. the first pass of radiation through the profile is processed. Not sure how to manage that....

thanks for thinking about this :)
-Rosie

**Comment by rosiealice** [comment]
I guess this is now concluded, in the presence of @adrifoster's successful testing of the two stream code?

**Comment by adrifoster** [comment]
Not sure ... it's still high, just not *as* high

https://webext.cgd.ucar.edu/I2000/FATES_global_4x5/lnd/_build_FATES/RadiationandEnergyCycle/Albedo/CERES/CERES.html

**Comment by rosiealice** [comment]
This is resolved by the twostream radiation scheme. 
https://github.com/NGEET/fates/pull/1036

but we perhaps need to make it default in the parameter file (and almost certainly should at this point) before we can close this out? @rgknox ?

**Comment by rgknox** [comment]
@rosiealice , while two-stream is stable, it still has some quirks that are giving restart testing issues.  As far as I can tell, its subtle things like reproducing history output on the first time-step of the restart, etc, but we need to get this squared away before we set it to default.

**Comment by rosiealice** [comment]
OK fair enough... Does that mean we don't expect it to pass restart tests if we change to having it as the default?

**Comment by rgknox** [comment]
yes @rosiealice , I've been meaning to prioritize this so that we can change 2-stream to default.  I've been focusing on things that affect calibration efforts answers as of late, but this is something I think is important to get done soon so that we can protect that code better.

**Comment by rosiealice** [comment]
Just noting that starting the NorESM spin up process we have high albedo biases that of course are likely messing up the climate, so maybe flagging this as perhaps a bigger issue...
